From c169c46ca4cbcd8c1093a492c92342016960e441 Mon Sep 17 00:00:00 2001
From: Hai Pham <hai.pham.ud@renesas.com>
Date: Tue, 9 Mar 2021 15:07:38 +0700
Subject: [PATCH 14/18] plat: rcar: common: disable R-Car PCIe fixup if V3U

Signed-off-by: Hai Pham <hai.pham.ud@renesas.com>
---
 plat/renesas/rcar/rcar_common.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plat/renesas/rcar/rcar_common.c b/plat/renesas/rcar/rcar_common.c
index e99df9b..715057a 100644
--- a/plat/renesas/rcar/rcar_common.c
+++ b/plat/renesas/rcar/rcar_common.c
@@ -61,6 +61,10 @@ void plat_ea_handler(unsigned int ea_reason, uint64_t syndrome, void *cookie,
 {
 	unsigned int fixed = 0;
 
+#if (RCAR_LSI == RCAR_V3U)
+	return;
+#endif
+
 	fixed |= rcar_pcie_fixup(0);
 	fixed |= rcar_pcie_fixup(1);
 
-- 
2.7.4

