--- a/qos-module/files/qos/drv/qos_reg.h
+++ b/qos-module/files/qos/drv/qos_reg.h
@@ -80,6 +80,8 @@
 #define MASTER_ID_MAX_H3_ES2		109
 #define MASTER_ID_MAX_M3_W		 99
 #define MASTER_ID_MAX_M3_N		114
+#define MASTER_ID_MAX_V3M		 41
+#define MASTER_ID_MAX_V3H		 99
 #define MASTER_ID_MAX_D3		114
 #define MASTER_ID_MAX			511
 
@@ -93,6 +95,8 @@
 #define R_CAR_H3			(0x00004F00U)
 #define R_CAR_M3_W			(0x00005200U)
 #define R_CAR_M3_N			(0x00005500U)
+#define R_CAR_V3M			(0x00005400U)
+#define R_CAR_V3H			(0x00005600U)
 #define R_CAR_D3			(0x00005800U)
 
 #define ES10				(0x00000000U)
--- a/qos-module/files/qos/drv/qos_core.c
+++ b/qos-module/files/qos/drv/qos_core.c
@@ -232,6 +232,21 @@
 				master_id_max = MASTER_ID_MAX_M3_N;
 				break;
 			}
+		} else if (device == R_CAR_V3M) {
+			switch (device_version) {
+			case ES10:
+			case ES20:
+			default:
+				master_id_max = MASTER_ID_MAX_V3M;
+				break;
+			}
+		} else if (device == R_CAR_V3H) {
+			switch (device_version) {
+			case ES10:
+			default:
+				master_id_max = MASTER_ID_MAX_V3H;
+				break;
+			}
 		} else if (device == R_CAR_D3) {
 			switch (device_version) {
 			case ES10:
