From 980b48bc279d46559465665e94e546f661d1d427 Mon Sep 17 00:00:00 2001
From: Grigory Kletsko <grigory.kletsko@cogentembedded.com>
Date: Fri, 17 Feb 2017 03:06:27 +0300
Subject: [PATCH 1/7] Allow to boot without input device

---
 libweston/libinput-seat.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/libweston/libinput-seat.c b/libweston/libinput-seat.c
index ac1e8e9..b7f601a 100644
--- a/libweston/libinput-seat.c
+++ b/libweston/libinput-seat.c
@@ -302,7 +302,6 @@ udev_input_enable(struct udev_input *input)
 			"\t- seats misconfigured "
 			"(Weston backend option 'seat', "
 			"udev device property ID_SEAT)\n");
-		return -1;
 	}
 
 	return 0;
-- 
2.7.4

