From e2fe50b628f660ef35c82b778c5d0c56b5c4487b Mon Sep 17 00:00:00 2001
From: Andrey Dolnikov <andrey.dolnikov@cogentembedded.com>
Date: Fri, 12 Jan 2018 18:25:33 +0300
Subject: [PATCH] spidev: Do not automatically bind to MSIOF spi (may be used
 as slave).

Signed-off-by: Andrey Dolnikov <andrey.dolnikov@cogentembedded.com>
---
 drivers/spi/spidev.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index 203d68c..cbd0b93 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -697,7 +697,6 @@ static struct class *spidev_class;
 static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "rohm,dh2228fv" },
 	{ .compatible = "lineartechnology,ltc2488" },
-	{ .compatible = "renesas,sh-msiof" },
 	{ .compatible = "spi-gpio" },
 	{},
 };
-- 
2.7.4

